### ISSUE TITLE
Meta data inheritance in stack of folders - Mixup metadata types

### ISSUE THREAD

**Issue by cnbhl** [ciur/papermerge#371]
**Description**
I created a folder and attach a meta data label with type "date". Next I created an additional folder inside the first folder, it inherits the meta data label, but the type is reset to "Freedom text" and can't be changed. 

For files inside the first folder the meta data type is correctly inherited. 

For files inside the second folder the but consists and the type can't be changed.

**Expected**
The subordinate folders and files inherit the meta data field types correctly. 

**Actual**
Folder 1 is created and a meta data field of type "date" is created. 
![38207605-folder1](https://user-images.githubusercontent.com/78079330/116385427-141c9b80-a819-11eb-9492-0e5c1a8d528f.jpg)

The meta data type attached to files inside this first folder are correctly set. (Btw: here you see a variation of  #223 on folders, I created "Folder 1.1" and "Folder 1.2", it's not a typo, will try to reproduce it.)
![38207605-document](https://user-images.githubusercontent.com/78079330/116385426-141c9b80-a819-11eb-9122-7e2743980257.jpg)

The one (or more) step inside folder 1.1 is created and inherits the meta data field, but the type is mixed up. And can't be changed.
![38207605-folder1 1](https://user-images.githubusercontent.com/78079330/116385422-13840500-a819-11eb-9c22-58b8136b40a8.jpg)

Files inside this 1.1 folder are also with the wrong meta data field.
![38207605-document1 1](https://user-images.githubusercontent.com/78079330/116385707-5a71fa80-a819-11eb-88d3-435002c5d5db.jpg)

**Info:**
 - OS: [e.g. Linux Debian 10, docker-compose]
 - Browser [Firefox 78.10.0]
-  Database [PostgreSQL 12.3, docker-compose]
 - Papermerge Version [2.0.1, docker-compose]

PS: outstanding project! I appreciate your work! 
I found this project after weeks of searching for a substitution of so called "open-source" ECM programms (who are actually attached to a company and in CE there is literally no functionality.)

**Comment by ciur** [comment]
Indeed, both folders Folder 1.1 and Folder 1.2 should have had inherited metadata of type "date" with format "dd.mm.yy" as specified by parent folder. 

@cnbhl, thanks for opening this issue with detailed description!
